src/tests/InputProblem.test.tsx

```diff
@@ -0,0 +1,184 @@
+import { render, screen, fireEvent, act } from '@testing-library/react'
+import { InputProblem } from '../components/InputProblem'
+import '@testing-library/jest-dom'
+
+test('InputProblem renders with content, textarea and button', async () => {
+  render(
+        <InputProblem
+      solvedCallback={() => { } }
+      exhaustedCallback={() => { } }
+      allowedRetryCallback={() => { } }
+      content={'Content'}
+      correctResponse={''}
+      encourageResponse={''}
+      retryLimit={0}
+      solution={' 5 '}
+      buttonText={'Submit'}
+      comparator={'integer'} />
+  )
+
+  screen.getByText('Content')
+  screen.getByRole('textbox')
+  expect(document.querySelector('textarea')).not.toBeNull()
+  expect(screen.getByRole('button').textContent).toBe('Submit')
+})
+
+test('Text InputProblem button click with correct answer should evaluate to correct', async () => {
+  const solvedHandler = jest.fn()
+  const exhaustedHandler = jest.fn()
+  const allowedRetryHandler = jest.fn()
+
+  render(
+          <InputProblem
+          solvedCallback={solvedHandler}
+          exhaustedCallback={exhaustedHandler}
+          allowedRetryCallback={allowedRetryHandler}
+          content={'Content'}
+          correctResponse={'Correct!'}
+          encourageResponse={''}
+          retryLimit={0}
+          solution={' Apple '}
+          buttonText={'Submit'}
+          comparator={'text'}
+          />
+  )
+  await act(async () => {
+    fireEvent.change(screen.getByRole('textbox'), { target: { value: 'Apple ' } })
+    screen.getByRole('button').click()
+  })
+  await screen.findByText('Correct!')
+  expect(solvedHandler).toBeCalledTimes(1)
+  expect(exhaustedHandler).toBeCalledTimes(0)
+  expect(allowedRetryHandler).toBeCalledTimes(0)
```

Comment:
Can we add assertions that the form is disabled when answered correctly to this test case?